Docs Update: Adding Upgrade guide to move from WalletConnect Modal to AppKit Basic

rohit-710 opened the pull request:
## Description

This PR creates a consolidated upgrade guide on Web to move from WalletConnect Modal to AppKit Basic. It covers Ethereum Provider, Universal Provider and Sign Client as well. The PR also has examples to each web library/programming language that developers can refer to. 

## Tests

- [x] - Ran the changes locally with Docusaurus. and confirmed that the changes appear as expected.
- [x] - Applied the corrections suggested by Cspell on the `.mdx` files with changes.

## Preview/s

- https://reown-docs-git-wcm-upgrade-reown-com.vercel.app/appkit/upgrade/wcm

rtomas commented:
we can add an image or/and a link to a demo after 

> WalletConnect Modal will soon be deprecated. Consequently, projects and developers need to upgrade to Reown AppKit Basic. The most basic version of Reown AppKit will contain the traditional WalletConnect Modal with a QR code that users can scan and connect their wallets to.

rtomas commented:
IMO will be great to add how to connect to Polkadot, Cosmos, etc ... with universal Protocol .. instead of Mainnet.
import { mainnet } from '@reown/appkit/networks'

rohit-710 commented:
> * IMO will be great to add how to connect to Polkadot, Cosmos, etc ... with universal Protocol .. instead of Mainnet.
>   import { mainnet } from '@reown/appkit/networks'
> * Also a link to this page can be added :
>   https://reown-docs-git-wcm-upgrade-reown-com.vercel.app/appkit/react/core/multichain?platform=basic
> * Maybe we should add all the titles in the top of the page.
> * `npm install` when new libraries is going to be used

I have created and added a demo video for this. I have also added installation steps and the titles on top of the page. There is also a link that redirects them to the multichain section of AppKit core. 

We can add Polkadot and Cosmos in the next iteration. The product team wanted to have this out today so they can send it to a few customers. But definitely a good point that we should showcase non-EVM chains like Polkadot and Cosmos.

rtomas reviewed (approved):
LGTM
